documentation/specs/Toast.md

```diff
@@ -0,0 +1,174 @@
+# `Toast` Component Specification
+
+## Overview
+
+The `Toast` component displays a brief non-disruptive message to the user as a result of an action taken. The displayed message is accompanied by an associated `status` icon and cannot be manually dismissed. **Not** to be used to display critical information such as a system failure, see the `Alert` component for such behavior.
+
+### Use Cases
+
+- Inform user of a completed action such as an account creation.
+- Inform user that an asynchronous process has been triggered based on their action.
+
+### Features
+
+- ARIA `role="status"`
+
+### Risks and Challenges
+
+- Unlike other components, `Toast` components are unique in that they may be quite opinionated with regards to how they display and the kind of control they give consumers.
+- Building for the shipper's experience at EasyPost according to internal specs while maintaining a pleasant DX for outside consumers.
+
+---
+
+## Design
+
+### API
+
+```typescript
+export type ToastStatus =
+  | "primary"
+  | "success"
+  | "secondary"
+  | "error"
+  | "warning";
+export type ToastProps = {
+  /**
+   * Toast status color
+   * @default "success"
+   */
+  status?: ToastStatus;
+  /**
+   * Toast content
+   */
+  children?: ReactNode;
+};
+```
```

Comment:
i know some libraries expose a `duration` property for the `Toast` and im open to adding it, but im having trouble envisioning a scenario where we add the property and have minimal state configuration within `Toast`

Reply:
i think we'll want to try to encapsulate the duration config within Easy UI, at least until we have a concrete use case for supporting multiple durations outside of the library